### DISCUSSION TITLE
SwaggerUI embed.FS not rendering when hit the endpoint

### DISCUSSION THREAD

**Discussion by dibrito** [MarioCarrion/todo-api-microservice-example#34] [Q&A] [answered]
Hey,

Is there anything missing for rendering the swagger UI download?
when I hit the endpoint: http://localhost:8080/openapi3.json
I get a 404.

What I'm doing basically  is:

`
	// mux router
	r := mux.NewRouter()

	// define handlers
	h := &customer.Handler{}

	// register handler to the mux
	h.Register(r)

	// swagger-ui
	fsys, _ := fs.Sub(content, "static")
	r.PathPrefix("/static/").Handler(http.StripPrefix("/static/", http.FileServer(http.FS(fsys))))

	address := "0.0.0.0:8080"
	// defines the server
	srv := &http.Server{
		Handler:      r,
		Addr:         address,
		ReadTimeout:       1 * time.Second,
		ReadHeaderTimeout: 1 * time.Second,
		WriteTimeout:      1 * time.Second,
		IdleTimeout:       1 * time.Second,
	}
`

**Comment by MarioCarrion** [comment, marked as answer]
@dibrito Apologies for the late reply it looks like I either missed the notification or I never got it.

If you're a getting a 404 when hitting the http://localhost:8080/openapi3.json endpoint then it means it wasn't added to the router. The code doing that is [here](https://github.com/MarioCarrion/todo-api-microservice-example/blob/bbecc092f7bb33fc60778a934f7e9f1db25147d6/cmd/rest-server/main.go#L226), which [registers the endpoints](https://github.com/MarioCarrion/todo-api-microservice-example/blob/bbecc092f7bb33fc60778a934f7e9f1db25147d6/internal/rest/open_api.go#L268-L284) with something like:

```go
	r.HandleFunc("/openapi3.json", func(w http.ResponseWriter, r *http.Request) {
		renderResponse(w, &swagger, http.StatusOK)
	}).Methods(http.MethodGet)
```
